pkg/datapath/iptables/iptables.go

```diff
@@ -1547,6 +1554,13 @@ func (m *Manager) installRules(state desiredState) error {
 		}
 	}
 
+	if m.haveTunnel {
+		// TODO only install these when needed, uninstall otherwise
+		if err := m.addCiliumNoTrackOverlayRules(); err != nil {
+			return fmt.Errorf("cannot install overlay rules: %w", err)
+		}
+	}
```

Comment:
@pippolo84 you might have more experience here - is cleaning up stale rules something we typically do? I don't see much evidence in `iptables.go` ...